modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -4405,9 +4425,10 @@ else if (log.isDebugEnabled())
                     }
                 }
 
-                IgniteNodeValidationResult err;
+                IgniteNodeValidationResult err = ensureJoinEnabled(node);
```

Comment:
Why do you decide to add the check here? There is an alternative - similar to checking node authentication. 

As I can see "validate" is two-factor process - `validateNode(node)`, `validateNode(node, data)`. The logic is described in `GridComponent#validateNode`. Looks it's a not obvious place for this check.